hudi-common/src/test/java/org/apache/hudi/common/model/TestPartialUpdateAvroPayload.java

```diff
@@ -0,0 +1,204 @@
+    record1.put("partition", "partition0");
+    record1.put("ts", 0L);
+    record1.put("_hoodie_is_deleted", false);
+    record1.put("city", "NY0");
+    record1.put("child", Collections.emptyList());
+
+    GenericRecord delRecord1 = new GenericData.Record(schema);
+    delRecord1.put("id", "2");
+    delRecord1.put("partition", "partition1");
+    delRecord1.put("ts", 1L);
+    delRecord1.put("_hoodie_is_deleted", true);
+    delRecord1.put("city", "NY0");
+    delRecord1.put("child", Collections.emptyList());
+
+    GenericRecord record2 = new GenericData.Record(schema);
+    record2.put("id", "1");
+    record2.put("partition", "partition0");
+    record2.put("ts", 0L);
+    record2.put("_hoodie_is_deleted", true);
+    record2.put("city", "NY0");
+    record2.put("child", Collections.emptyList());
+
+    PartialUpdateAvroPayload payload1 = new PartialUpdateAvroPayload(record1, 0);
+    PartialUpdateAvroPayload payload2 = new PartialUpdateAvroPayload(delRecord1, 1);
+
+    assertEquals(payload1.preCombine(payload2), payload2);
+    assertEquals(payload2.preCombine(payload1), payload2);
+
+    assertEquals(record1, payload1.getInsertValue(schema).get());
+    assertFalse(payload2.getInsertValue(schema).isPresent());
+
+    Properties properties = new Properties();
+    properties.put(HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY, "ts");
+    assertEquals(payload1.combineAndGetUpdateValue(delRecord1, schema, properties), Option.empty());
+    assertFalse(payload2.combineAndGetUpdateValue(record1, schema, properties).isPresent());
+  }
+
+  @Test
+  public void testNullColumn() throws IOException {
+    Schema avroSchema = Schema.createRecord(Arrays.asList(
+        new Schema.Field("id", Schema.createUnion(Schema.create(Schema.Type.STRING), Schema.create(Schema.Type.NULL)), "", JsonProperties.NULL_VALUE),
+        new Schema.Field("name", Schema.createUnion(Schema.create(Schema.Type.STRING), Schema.create(Schema.Type.NULL)), "", JsonProperties.NULL_VALUE),
+        new Schema.Field("age", Schema.createUnion(Schema.create(Schema.Type.STRING), Schema.create(Schema.Type.NULL)), "", JsonProperties.NULL_VALUE),
+        new Schema.Field("job", Schema.createUnion(Schema.create(Schema.Type.STRING), Schema.create(Schema.Type.NULL)), "", JsonProperties.NULL_VALUE)
+    ));
+    GenericRecord record1 = new GenericData.Record(avroSchema);
+    record1.put("id", "1");
+    record1.put("name", "aa");
+    record1.put("age", "1");
+    record1.put("job", "1");
+
+    GenericRecord record2 = new GenericData.Record(avroSchema);
+    record2.put("id", "1");
+    record2.put("name", "bb");
+    record2.put("age", "2");
+    record2.put("job", null);
+
+    GenericRecord record3 = new GenericData.Record(avroSchema);
+    record3.put("id", "1");
+    record3.put("name", "bb");
+    record3.put("age", "2");
+    record3.put("job", "1");
+
+    OverwriteNonDefaultsWithLatestAvroPayload payload2 = new OverwriteNonDefaultsWithLatestAvroPayload(record2, 1);
```

Comment:
This overlaps w/ the first test, doesn't it? Do we need it separately?

Reply:
this test seems not need to exist in this version